src/vma/dev/net_device_val.cpp

```diff
@@ -1080,17 +1085,21 @@ resource_allocation_key* net_device_val::ring_key_redirection_release(resource_a
 		return ret_key;
 	}
 
-	nd_logdbg("release redirecting key=%s (ref-count:%d) to key=%s", key->to_str(),
-		m_h_ring_key_redirection_map[key].second,
-		m_h_ring_key_redirection_map[key].first->to_str());
 	ret_key = m_h_ring_key_redirection_map[key].first;
-	if (--m_h_ring_key_redirection_map[key].second == 0) {
+
+	return ret_key;
+}
+
+void net_device_val::ring_key_redirection_release(resource_allocation_key *key)
```

Comment:
Please add a check which verifies if key exist in the map.
Currently, because you are the only one which uses this function, It does not matter. But it can solve problems in the future when/if this function will be used from other locations.